pkg/controller/common/webhook/custom_resource_deletion.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
I also worry a bit about the location of this file? Any better locations for this? Maybe a new package?